events/UserDataCollectionEvent.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\legal\events;
+
+use humhub\modules\comment\models\Comment;
+use humhub\modules\file\libs\FileHelper;
+use humhub\modules\file\models\File;
+use humhub\modules\legal\services\ExportService;
+use humhub\modules\like\models\Like;
+use humhub\modules\post\models\Post;
+use humhub\modules\rest\definitions\CommentDefinitions;
+use humhub\modules\rest\definitions\FileDefinitions;
+use humhub\modules\rest\definitions\LikeDefinitions;
+use humhub\modules\rest\definitions\PostDefinitions;
+use humhub\modules\rest\definitions\UserDefinitions;
+use humhub\modules\user\events\UserEvent;
+use Yii;
+use yii\base\Exception;
+use ZipArchive;
+
+class UserDataCollectionEvent extends UserEvent
+{
+    /**
+     * Array with export data:
+     *  - Key is a category name which will be used to name JSON file in ZIP archive
+     *  - Value is an array with any data
+     * @var array $exportData
+     */
+    public array $exportData = [];
+
+    /**
+     * Array with export files:
+     *  - Key is a file name which will be used in ZIP archive
+     *  - Value is a file path where it is really located on the server disk
+     * @var array $exportFiles
+     */
+    public array $exportFiles = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        $this->addExportData('user', UserDefinitions::getUser($this->user));
```

Comment:
@yurabakhtin Can we add the data of core here via the Rest module? In the same way as the other modules?

To cover the core data completely in the package, we need to create some more definitions. I think this is easier to do in the REST module.

Reply:
@luke- Yes, good idea, done https://github.com/humhub/rest/pull/177 and https://github.com/humhub/legal/pull/77/commits/de4833ddb1ccb1625b74dc5236eb22f2a3071c72.